paws_collector.js

```diff
@@ -780,6 +808,79 @@ class PawsCollector extends AlAwsCollector {
         });
     }
 
+    /**
+     * Get hash of message 
+     * @param {*} message 
+     * @returns 
+     */
+    getHash(message) {
+        const hashValue = crypto.createHash('sha256').update(JSON.stringify(message, Object.keys(message).sort())).digest('hex');
+        return hashValue;
+    }
+
+   /**
+    * 
+    * @param {*} logs 
+    * @param {*} paramName :Uniquely identified parameter key 
+    * @param {*} callback 
+    * @returns 
+    */
+    removeDuplicatedItem(logs, paramName, asyncCallback) {
+        let collector = this;
+        const ddb = new AWS.DynamoDB();
+        let uniqueLogs = [];
+        var promises = [];
+        let duplicateCount = 0;
+        logs.forEach(record => {
+            const messageHashId = collector.getHash(record);
+            const params = {
+                Item: {
+                    Id: { S: record[`${paramName}`] },
+                    CollectorId: { S: collector._collectorId },
+                    MsgHashId: { S: messageHashId },
+                    // setting DDB time to life. This is set to cover 24hr duplication window
+                    ExpireDate: { N: moment().add(1, 'days').unix().toString() }
+                },
+                TableName: collector._pawsDeDupLogsDdbTableName,
+                ConditionExpression: '(attribute_not_exists(Id) AND attribute_not_exists(MsgHashId)) OR  (Id = :id AND MsgHashId <> :msg_hash_id AND CollectorId = :collector_id)', // Check if item with same Id already exists and id and message is identical
+                ExpressionAttributeValues: {
+                    ':id': { S: record[`${paramName}`] }, ':msg_hash_id': { S: messageHashId }, ':collector_id': { S: collector._collectorId }
+                }
+            };
+            let promise = new Promise((resolve, reject) => {
+                ddb.putItem(params, (err, res) => {
+                    if (err) {
+                        if (err.code === 'ConditionalCheckFailedException') {
+                            duplicateCount++;
+                            return resolve(null);
+                        } else {
+                            AlLogger.warn('PAWS000404 Error storing event in DynamoDB:', err);
+                            return reject(err);
+                        }
+                    }
+                    else {
+                        uniqueLogs.push(record);
+                        return resolve(null);
+                    }
+                });
+            });
+            promises.push(promise);
+        });
+
+        Promise.all(promises).then(result => {
+            if (duplicateCount > 0) {
+                collector.reportDuplicateLogCount(duplicateCount, (err)=>{
+                    if(err)
+                    AlLogger.warn(`PAWS000405 error from custom cloud watch metrics ${JSON.stringify(err)}`);
```

Comment:
this syntax scares me 😱 

could you please enclose the body of this expression in curly braces?
```javascript
if(err) {
  // ...
}
```

Reply:
sure